### PR TITLE
Test docs

### PR DESCRIPTION
**Pull request by mfalt**
* Replaces https://github.com/JuliaControl/ControlSystems.jl/pull/293 because of trouble merging
* Updates tests to require plots to not fail and docs-tests to pass.
* Have copy of code to generate test plots locally in test
* Updated so docs-tests are run on all examples in documentation, including plots 
* Removed duplicate code for generating documentation plots
* Temporary fix to `jldoctest` to be able to hide lines using # hide
* Other smaller fixes

### CODE REVIEW

**Comment by mfalt** [comment]
If tests pass then i think this is ready to go. I looked at the docs locally, and saw that we need to fix the title in stabreagionPID. Opened a new issue for that.

File: docs/src/examples/example.md [hunk 1]
```diff
@@ -118,7 +133,12 @@ R,S,T = rstc(B⁺,B⁻,A,Bm,Am,Ao,AR) # Calculate the 2-DOF controller polynomia
 Gcl = tf(conv(B,T),zpconv(A,R,B,S)) # Form the closed loop polynomial from reference to output, the closed-loop characteristic polynomial is AR + BS, the function zpconv takes care of the polynomial multiplication and makes sure the coefficient vectores are of equal length
 
 stepplot([P,Gcl]) # Visualize the open and closed loop responses.
+save_docs_plot("ppstepplot.svg") # hide
```
Comment (baggepinnen) [inline docs/src/examples/example.md:136]: I learned the other day that plots appear in the docs automatically now if they are the output of a block. No need to change this now, but nice to remember for the future.

File: docs/src/examples/example.md [hunk 2]
```diff
@@ -143,24 +170,42 @@ f3 = stabregionPID(P3,exp10.(range(-5, stop=0, length=1000)))
...
 ki = ws.^2
-pidplots(P,:nyquist,:gof;kps=kp,kis=ki, ω= exp10.(range(-2, stop=2, length=500))) # Request Nyquist and Gang-of-four plots (more plots are available, see ?pidplots )
+
+ω = exp10.(range(-3, stop=2, length=500))
+pidplots(P,:nyquist;kps=kp,kis=ki,ω=ω, ylims=(-2,2), xlims=(-3,3))
+save_docs_plot("pidplotsnyquist1.svg") # hide
+pidplots(P,:gof;kps=kp,kis=ki,ω=ω,legend=false)
+# You can also request both Nyquist and Gang-of-four plots (more plots are available, see ?pidplots ):
+# pidplots(P,:nyquist,:gof;kps=kp,kis=ki,ω=ω);
+save_docs_plot("pidplotsgof1.svg"); # hide
```
Comment (baggepinnen) [inline docs/src/examples/example.md:188]:
```suggestion
ω = exp10.(range(-3, stop = 2, length = 500))
pidplots(
    P,
    :nyquist;
    kps = kp,
    kis = ki,
    ω = ω,
    ylims = (-2, 2),
    xlims = (-3, 3),
)
save_docs_plot("pidplotsnyquist1.svg") # hide
pidplots(P, :gof; kps = kp, kis = ki, ω = ω, legend = false)
# You can also request both Nyquist and Gang-of-four plots (more plots are available, see ?pidplots ):
# pidplots(P,:nyquist,:gof;kps=kp,kis=ki,ω=ω);
save_docs_plot("pidplotsgof1.svg"); # hide
```
make it easier to parse

**Review by baggepinnen — APPROVED** [review]
I didn't look super carefully since this PR is so large, but it looks good to me :)

File: docs/src/examples/example.md (hunk 1 above)
Comment (mfalt) [reply docs/src/examples/example.md:136]: Great to know

**Comment by mfalt** [comment]
@JuliaControlBot test-plots

**Pull request merged**